src/test/scala/retryFunction/RetryFunctionSpec.scala

```diff
@@ -1,33 +1,50 @@
 package retryFunction
 
+import org.scalatest.GivenWhenThen
 import org.scalatest.flatspec.AnyFlatSpec
 import org.scalatest.matchers.should.Matchers
 
-class RetryFunctionSpec extends AnyFlatSpec with Matchers {
+class RetryFunctionSpec extends AnyFlatSpec with Matchers with GivenWhenThen{
 
   "RetryFunction.retry" should "return '3' without additional retries" in {
+    Given("input block function '() => 1 + 2' and accept function 'res => res > 0'. Number of potential retries: 3 (100, 1000, 2000)")
+    When("retry function is executed")
+    Then("result of execution should be 3")
     RetryFunction.retry[Int](
       block = () => 1 + 2,
       accept = res => res > 0,
       retries = List(100, 1000, 2000)
     ) should equal(3)
   }
 
-  it should "do only 1 retry" in {
+  it should "do only 1 retry if accept function return true after it" in {
+    Given("util variable of Utility type. Only one method tempIncrement is used in this test - for incrementation of initial temp value")
+    Given("input block function '() => util.tempIncrement()' and accept function 'res => res == 1'. Number of potential retries: 3 (100, 100, 100)")
```

Comment:
Please, use `And` function instead of repeating `Given`.

Reply:
All second When or Then were changed to Add